### PR TITLE
Feature/6738 messenger css customization

### PR DESCRIPTION
**Pull request by Cesarcs7**
The codesandbox Demo link will go after the facebook messenger section, after "See it in action" we have to wait till this PR is in master gemerged so I can link the demo to the master repo and be able to use the new CSS-classes.

I remove some duplicated code from the old CSS-customization documentation, also added the new one for the Facebook messenger elements. I will hang a CSS file in Azure so you can test the new results, also please connect to this endpoint where you can easily test all Facebook messenger templates.

https://endpoint-demo.cognigy.ai/597fc5500b1b6fad9a4a8f1a2f1fa9940af9d3b8a70def2c17a236105bb34d38

### CODE REVIEW

File: docs/css-customization.md [hunk 1]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
 }
 
 ```
+## Facebook Messenger messages customization
```
[the remark this answers is not in the document]
Comment (dshire) [reply docs/css-customization.md]: If we change the naming to something not related to Messenger, we also have to rename the classNames.

File: docs/css-customization.md [hunk 2]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+  border:2px solid  #fffffe;
+  border-radius: 0;
+  text-align: center;
+  background: #e3f6f5;
+  color: #272343;
+  animation: "Some animation" ;
+}
+
+```
+* *messenger-quick-reply-replies*
```
Comment (dshire) [inline docs/css-customization.md]: not sure about the naming here: Isn't this the actual container? If so, then "messenger-quick-reply-reply-container" could be called e.g. "webchat-quick-reply-root"

File: docs/css-customization.md [hunk 3]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+* *messenger-quick-reply-replies*  
+This is the element holding all Quick Replies, you can change the way they are shown. Like a column.
+```CSS
+[data-cognigy-webchat-root] .messenger-quick-reply-replies {
+  display: flex;
+  flex-direction: column;
+}
+
+```
+* *messenger-quick-reply-reply-container*
```
Comment (dshire) [inline docs/css-customization.md]: see comment line 317

File: docs/css-customization.md [hunk 4]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+  .messenger-quick-reply-reply-text {
+  color: yellow;
+}
+
+```
+
+### Buttons
+
+* *messenger-buttons*  
+This class contents the Buttons element.
```
Comment (dshire) [inline docs/css-customization.md]: Did you mean "contains" ?

File: docs/css-customization.md [hunk 5]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+```CSS
+[data-cognigy-webchat-root] .messenger-buttons {
+  border-radius: 0;
+  animation: "Some animation" ;
+}
+
+```
+* *messenger-buttons-header*  
+The container and header of the buttons, you could modify the text position and style.
```
Comment (dshire) [inline docs/css-customization.md]: "can" modify

File: docs/css-customization.md [hunk 6]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+  animation-iteration-count: 1;
+  font-weight: bold;
+}
+
+```
+
+### Image Gallery
+
+* *messenger-carousel-container*  
+The element that contains the card from a Gallery, here you can increase it's size for example.
```
Comment (dshire) [inline docs/css-customization.md]: This is the card element itself or not? Then it should not be "The element that contains the card from a Gallery", but  "The card element of a gallery"

File: docs/css-customization.md [hunk 7]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+* *messenger-carousel-title*  
+The title of the card.
+```CSS
+[data-cognigy-webchat-root] .messenger-carousel-title {
+    color:blue;
+}
+
+```
+* *messenger-carousel-subtitle*  
+The title of the card.
```
Comment (dshire) [inline docs/css-customization.md]: *subtitle

File: docs/css-customization.md [hunk 8]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+}
+
+[data-cognigy-webchat-root] .messenger-media-audio {
+}
+```
+
+### List
+
+* *messenger-list-container*  
+The container element of the List, we could remove the border-radius that comes with it.
```
Comment (dshire) [inline docs/css-customization.md]: you can remove

File: docs/css-customization.md [hunk 9]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+```
+* *messenger-list-element*  
+The element that contains a single list element
+```CSS
+[data-cognigy-webchat-root] .messenger-list-element {
+  background: #e3f6f5;
+}
+```
+* *messenger-list-element-title*  
+The title in every list element.
```
Comment (dshire) [inline docs/css-customization.md]: of every

File: docs/css-customization.md [hunk 10]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+```CSS
+[data-cognigy-webchat-root] .messenger-list-element-title {
+  color: yellow;
+}
+```
+* *messenger-list-element-subtitle*  
+The subtitle in every list element.
```
Comment (dshire) [inline docs/css-customization.md]: of every

File: src/plugins/messenger/MessengerPreview/components/MessengerButtonTemplate/MessengerButtonTemplate.tsx [hunk 11]
```diff
@@ -1,47 +1,49 @@
-import { IFBMButtonTemplatePayload } from '../../interfaces/ButtonTemplatePayload.interface';
```
Comment (dshire) [inline src/plugins/messenger/MessengerPreview/components/MessengerButtonTemplate/MessengerButtonTemplate.tsx]: please revert undesired styling changes. E.g.: This file now has tab stops of 2 spaces, we always use 4

File: src/plugins/messenger/MessengerPreview/components/MessengerGenericTemplate/MessengerGenericTemplate.tsx [hunk 12]
```diff
@@ -1,135 +1,155 @@
-import { IFBMGenericTemplatePayload, IFBMGenericTemplateElement } from '../../interfaces/GenericTemplatePayload.interface';
```
Comment (dshire) [inline src/plugins/messenger/MessengerPreview/components/MessengerGenericTemplate/MessengerGenericTemplate.tsx]: please revert undesired styling changes. E.g.: This file now has tab stops of 2 spaces, we always use 4

File: src/plugins/messenger/MessengerPreview/components/MessengerMediaTemplate/MessengerMediaTemplate.tsx [hunk 13]
```diff
@@ -1,94 +1,95 @@
-import ReactPlayer from 'react-player';
```
Comment (dshire) [inline src/plugins/messenger/MessengerPreview/components/MessengerMediaTemplate/MessengerMediaTemplate.tsx]: please revert undesired styling changes. E.g.: This file now has tab stops of 2 spaces, we always use 4

File: src/plugins/messenger/MessengerPreview/components/MessengerTextWithQuickReplies/MessengerTextWithQuickReplies.tsx [hunk 14]
```diff
@@ -1,104 +1,112 @@
-import { IFBMRegularMessage } from '../../interfaces/Message.interface';
```
Comment (dshire) [inline src/plugins/messenger/MessengerPreview/components/MessengerTextWithQuickReplies/MessengerTextWithQuickReplies.tsx]: please revert undesired styling changes. E.g.: This file now has tab stops of 2 spaces, we always use 4

File: docs/css-customization.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md [hunk 15]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
 }
 
 ```
+## Facebook Messenger messages customization
+
+Our Webchat ships with "Facebook-Messenger-like" message templates out of the box (Quick replies, Buttons, Image Galleries, Media, Lists) these elements can be also customized to meet design guidelines even more!
```
[the remark this answers is not in the document]
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md [hunk 16]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+
+Here are the classes that you can use to modify it:
+
+See it in action:
```
[the remark this answers is not in the document]
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md [hunk 17]
```diff
@@ -373,3 +281,248 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+
+
+### Quick Replies
+
+* *messenger-quick-reply*  
+This class contains the whole Quick Reply element, you can modify it's size for example.
```
[the remark this answers is not in the document]
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 2 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 3 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 4 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: deleted

File: docs/css-customization.md (hunk 5 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 6 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 7 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 8 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 9 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 10 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: src/plugins/messenger/MessengerPreview/components/MessengerButtonTemplate/MessengerButtonTemplate.tsx (hunk 11 above)
Comment (Cesarcs7) [reply src/plugins/messenger/MessengerPreview/components/MessengerButtonTemplate/MessengerButtonTemplate.tsx]: fixed

File: src/plugins/messenger/MessengerPreview/components/MessengerGenericTemplate/MessengerGenericTemplate.tsx (hunk 12 above)
Comment (Cesarcs7) [reply src/plugins/messenger/MessengerPreview/components/MessengerGenericTemplate/MessengerGenericTemplate.tsx]: fixed

File: src/plugins/messenger/MessengerPreview/components/MessengerMediaTemplate/MessengerMediaTemplate.tsx (hunk 13 above)
Comment (Cesarcs7) [reply src/plugins/messenger/MessengerPreview/components/MessengerMediaTemplate/MessengerMediaTemplate.tsx]: fixed

File: src/plugins/messenger/MessengerPreview/components/MessengerTextWithQuickReplies/MessengerTextWithQuickReplies.tsx (hunk 14 above)
Comment (Cesarcs7) [reply src/plugins/messenger/MessengerPreview/components/MessengerTextWithQuickReplies/MessengerTextWithQuickReplies.tsx]: fixed

File: docs/css-customization.md [hunk 18]
```diff
@@ -373,3 +281,236 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+  border-radius: 0;
+  text-align: center;
+  background: #e3f6f5;
+  color: #272343;
+  animation: "Some animation" ;
+}
+
+```
+* *webchat-quick-reply-template-replies-container*  
+This is the element holding all Quick Replies, you can change the way they are shown. Like a column.
```
Comment (dshire) [inline docs/css-customization.md]: "Like a column." is not a proper sentence. You could write "This is the element holding all Quick Replies, you can change the way they are shown, e.g. as a column.

File: docs/css-customization.md [hunk 19]
```diff
@@ -373,3 +281,236 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+  animation-iteration-count: 1;
+  font-weight: bold;
+}
+
+```
+
+### Image Gallery
+
+* *webchat-carousel-template-root*  
+The card element a Gallery, here you can increase its size for example.
```
Comment (dshire) [inline docs/css-customization.md]: missing a word here

File: docs/css-customization.md [hunk 20]
```diff
@@ -373,3 +281,236 @@ This is the button to open the Webchat, if you want to modify its color you have
...
+
+[data-cognigy-webchat-root] .webchat-carousel-template-button:hover {
+  background: #cfb000;
+}
+```
+
+### Media
+
+* *webchat-media-template*  
+The media element can be of three forms, Image, Video or Audio. You can modify its container or put some animations on it
```
Comment (dshire) [inline docs/css-customization.md]: The media element can be of three forms: Image, Video or Audio. You can modify its container or put some animations on it.

File: docs/css-customization.md (hunk 18 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 19 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

File: docs/css-customization.md (hunk 20 above)
Comment (Cesarcs7) [reply docs/css-customization.md]: fixed

**Pull request merged**